concurrency_test.go

```diff
@@ -26,3 +26,47 @@ func TestAsync(t *testing.T) {
 		is.Fail("Async should not block")
 	}
 }
+
+func TestAwait(t *testing.T) {
+	is := assert.New(t)
+
+	aAsync := Async(func() int { return 1 })
```

Comment:
It might be best not to use `Async` for this test, that way, if Async is broken, this test would still pass.

You could just create channels in the main thread and send a value to it before to call AwaitAllX.